### ISSUE TITLE
Remove single line annotation comments from #include directives

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/omr#3521]
In the compiler component many of the `#include` pre-processor directives have a single line annotation describing what feature is being included from that file.  For example,
```
#include "codegen/CodeGenerator.hpp"           // for CodeGenerator
#include "codegen/FrontEnd.hpp"                // for TR_FrontEnd, etc
#include "codegen/RecognizedMethods.hpp"       // for RecognizedMethod, etc
#include "compile/Compilation.hpp"             // for Compilation, etc
#include "compile/CompilationTypes.hpp"        // for TR_Hotness
#include "compile/Method.hpp"                  // for TR_Method
```
These annotations were produced from a tool (IWYU) before the code was contributed to open source and in many places have gone stale or the formatting has degraded over time.  They are not particularly useful anyway when there are multiple reasons for including the files.

This issue tracks the work to remove such comments from all source and header files.

**Comment by Rajlaxmi** [comment]
Hi,
I would like to pick this up.

**Comment by Rajlaxmi** [comment]
I have used sed in .hpp and .cpp files.
Expression: 's/\(^\#include[ \t]*[<"].*[>"]\)[ \t]*\/\/[ \t]* for .*/\1/g'

**Comment by 0xdaryl** [comment]
Fixed in #3522.
